cpp/src/arrow/flight/client.h

```diff
@@ -94,8 +94,6 @@ class ARROW_FLIGHT_EXPORT FlightWriteSizeStatusDetail : public arrow::StatusDeta
 
 class ARROW_FLIGHT_EXPORT FlightClientOptions {
  public:
-  FlightClientOptions();
```

Comment:
I declared default constructor as private and find it's called implicitly at line https://github.com/apache/arrow/blob/master/cpp/src/arrow/flight/client.cc#L1243, the second augment `{}`.

Default constructor creates invalid object now, should not be accidentally called.
Maybe add a constructor with explicit parameters?

Reply:
Ok, since this is a footgun, I've changed these options to behave like the IPC options: while there's a Defaults(), the default values are all set as default field values and the default constructor is left as-is.